src/xb-builder-source-ctx.c

```diff
@@ -98,6 +103,20 @@ xb_builder_source_ctx_get_bytes (XbBuilderSourceCtx *self,
 	g_return_val_if_fail (XB_IS_BUILDER_SOURCE_CTX (self), NULL);
 	g_return_val_if_fail (cancellable == NULL || G_IS_CANCELLABLE (cancellable), NULL);
 	g_return_val_if_fail (error == NULL || *error == NULL, NULL);
+
+	/* Try mmap()ing the file first, as that avoids buffer allocation.
+	 * Note that this imposes the restriction that the backing file must not
+	 * be modified during the lifetime of the returned #GBytes. */
+	if (priv->file != NULL) {
+		g_autoptr(GMappedFile) mapped_file = NULL;
+		g_autoptr(GError) error_local = NULL;
+		const gchar *filename = g_file_peek_path (priv->file);
+
+		mapped_file = g_mapped_file_new (filename, FALSE, &error_local);
```

Comment:
I think nuke error_local not, i.e. -> NULL.

Reply:
Oops, yes!